pkg/cmd/importer/import.go

```diff
@@ -71,70 +66,68 @@ func NewOptions(ioStreams streams.IO) *Options {
 }
 
 // Complete fills out default values before running
-func (o *Options) Complete() error {
-	if o.PostgresURI == "" {
-		return fmt.Errorf("must provide postgres uri or dsn")
+func (o *Options) Complete(ctx context.Context, args []string) error {
+	if len(args) == 1 {
+		o.PostgresURI = args[0]
 	}
 
-	cfg, err := pgxpool.ParseConfig(o.PostgresURI)
-	if err != nil {
+	if err := o.PostgresOptions.Complete(); err != nil {
 		return err
 	}
-	o.poolConfig = cfg
 
-	if len(o.MappingFile) == 0 && o.config == nil {
-		return fmt.Errorf("no mapping config file set")
+	if err := o.SpiceDBOptions.Complete(o.DryRun); err != nil {
+		return err
 	}
 
-	if o.config == nil {
-		mapFileContents, err := os.ReadFile(o.MappingFile)
-		if err != nil {
-			return err
-		}
-		if err := json.Unmarshal(mapFileContents, &o.config); err != nil {
-			return err
-		}
+	if err := o.ConfigOptions.Complete(ctx, o.ReplogConfig, o.IO); err != nil {
+		return err
 	}
 
 	if o.DryRun {
-		o.writer = write.NewDryRunRelationshipWriter()
-		return nil
-	}
-	if o.DryRun {
-		o.writer = write.NewDryRunRelationshipWriter()
+		o.RelationshipWriter = write.NewDryRunRelationshipWriter()
 		return nil
 	}
 
-	if o.SpiceDBEndpoint == "" {
-		return fmt.Errorf("must provide spicedb uri")
-	}
-	grpcOpts := make([]grpc.DialOption, 0)
-	if o.SpiceDBInsecure == true {
-		grpcOpts = append(grpcOpts, grpc.WithInsecure())
-	}
-	if o.SpiceDBToken != "" && o.SpiceDBInsecure {
-		grpcOpts = append(grpcOpts, grpcutil.WithInsecureBearerToken(o.SpiceDBToken))
-	}
-	if o.SpiceDBToken != "" && !o.SpiceDBInsecure {
-		grpcOpts = append(grpcOpts, grpcutil.WithBearerToken(o.SpiceDBToken))
-	}
-	client, err := authzed.NewClient(o.SpiceDBEndpoint, grpcOpts...)
-	if err != nil {
-		return err
-	}
-	o.writer = write.NewRelationshipWriter(client)
+	o.RelationshipWriter = write.NewRelationshipWriter(o.Client)
 	return nil
 }
 
 // Run runs the command configured by Options.
 func (o *Options) Run(ctx context.Context) error {
-	log.Info().EmbedObject(util.LoggedConnConfig{ConnConfig: o.poolConfig.ConnConfig}).Msg("connecting to postgres")
+	// TODO: schema RelationshipWriter
```

Comment:
this probably needs to be a whole gh issue rather than just a TODO right?

Reply:
I just went ahead and implemented it; I had meant to resolve that before pushing and just forgot